src/rgw/rgw_sal_motr.cc

```diff
@@ -1885,6 +1884,121 @@ int MotrObject::delete_obj_aio(const DoutPrefixProvider* dpp, RGWObjState* astat
   return 0;
 }
 
+int MotrCopyObj_CB::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len)
+{
+  int rc = 0;
+  ldpp_dout(m_dpp, 20) << "Offset = " << bl_ofs << " Length = " << bl_len << dendl;
+  
+  rc = m_dst_writer->process(std::move(bl), bl_ofs);
```

Comment:
we should not pass bl_ofs here, this is local offset of bl.
in `MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)` , `offset` is offset of the entire object and in `data`, data is always present from 0th index, which is not the case with `bl`
We need to create a local `data`, copy `bl` starting from bl_ofs to bl_len.
You can refer `RGWRESTStreamOutCB::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len)` function where this is handled.

Update the `this->offfset `
`this->offfset += bl_len;`

Then pass this->offfset in the process function.

Reply:
made some changes @sjain09. Please review.